### ISSUE TITLE
Test babel-preset-yoshi

### ISSUE THREAD

**Issue by ronami** [wix/yoshi#1221]
### Summary

Currently, we have no tests for `babel-preset-yoshi`. If or when things break, we'll know about it only after we release or merge into `master`. We did in fact break recently with legacy decorators.

We can add tests fairly easy to every supported feature by adding use cases (features) to our kitchensink tests (https://github.com/wix/yoshi/tree/master/test/projects) and making sure they render correctly in the browser.

**Comment by yanivefraim** [comment]
I think we can close this for now. @ronami, wdyt?

**Comment by ranyitz** [comment]
@yanivefraim I think that there are more things that we would like to test, like the fact that the preset [outputs different targets according to the environment](https://github.com/wix/yoshi/blob/master/packages/babel-preset-yoshi/index.js#L44-L48) (`NODE_ENV`/`BABEL_ENV`) 

Or that in `test`/`development` we don't transpile `object-rest-spread` but just [understand the syntax](https://github.com/wix/yoshi/blob/master/packages/babel-preset-yoshi/index.js#L94-L95) which in `production` we do.

We can also test that the "ignoreReact" option actually works and [disable the propTypes removal](https://github.com/wix/yoshi/blob/master/packages/babel-preset-yoshi/index.js#L102) on production projects.

We can close this issue and open another one that specifies some tests that we would like to add (Just like I added above)

**Comment by yanivefraim** [comment]
Closing for #1904 and #1886
